src/main/kotlin/io/github/gradlenexus/publishplugin/NexusPublishPlugin.kt

```diff
@@ -55,16 +56,23 @@ class NexusPublishPlugin : Plugin<Project> {
         val extension = project.extensions.create<NexusPublishExtension>(NexusPublishExtension.NAME, project)
 
         extension.repositories.all {
+            val stagingRepositoryId = project.objects.property<String>()
             project.tasks.register("publishTo${capitalizedName()}") {
                 description = "Publishes all Maven publications produced by this project to the '${this@all.name}' Nexus repository."
                 group = PublishingPlugin.PUBLISH_TASK_GROUP
             }
             project.tasks
-                    .register<InitializeNexusStagingRepository>("initialize${capitalizedName()}StagingRepository", project.objects, extension, this, serverUrlToStagingRepoUrl)
+                    .register<InitializeNexusStagingRepository>("initialize${capitalizedName()}StagingRepository", project.objects, extension, this, serverUrlToStagingRepoUrl, { id: String -> stagingRepositoryId.set(id) })
```

Comment:
I like the trick to use a function to set (and later use) dangling property in not directly bound contexts. It greatly reduced complexity of the workaround on finalized properties in Gradle.